### ISSUE TITLE
Allow the input to be modified via options

### ISSUE THREAD

**Issue by sgilroy** [matteofigus/grunt-api-benchmark#5]
For API endpoints which require authentication, how can the appropriate headers be added to the request?

The solution I came up with (see the commit to my fork https://github.com/TwineHealth/grunt-api-benchmark/commit/dfd10d3354ed00c154e170b7621c33157041f062) was to add options.prepareInput.

In my Gruntfile, I prepare for the benchmark by first creating an appropriate access token and inject the prepareFunction into the config which will add an Authorization header to every endpoint. It looks something like this:
```
    grunt.registerTask('prepare-api-benchmark', function() {
        var done = this.async();
        
        Promise.cast(User.findOne({ email_address: 'special-user@example.com' }).exec()).then(function(user) {
            return Factory.create('access_token', { _user: user.id }).then(function (token) {
                grunt.config.set('api_benchmark.my_api.options.prepareInput', function(input) {
                    _.forEach(input.endpoints, function(endpoint) {
                        if (_.isObject(endpoint)) {
                            endpoint.headers = endpoint.headers || {};
                            endpoint.headers["Authorization"] = "Bearer " + token.token;
                        }
                    });
                });
                done();
            });
        });
    });

```

Alternatives:
 1. Ensure that all API's can be used without authentication (perhaps by configuring a non-production environment differently)
 2. Dynamically generate the input JSON file (perhaps via a template) and use the generated input file each time the benchmark is run. This has the downside of an additional build artifact and more file access than an in-memory solution. It also puts more burden on the user (vs the grunt-api-benchmark library).
 3. Don't use grunt-api-benchmark and instead use api-benchmark directly (where endpoints can contain functions and are specified in JavaScript config blocks instead of a JSON file).

**Comment by matteofigus** [comment]
Perhaps we can allow the configuration to be dynamicly passed in instead of just via JSON path. This should allow you to do this and to be very flexible (in the same way you can be using directly `api-benchmark`)

**Comment by sgilroy** [comment]
@matteofigus Yes, I agree that would accomplish what I need and would be more flexible. I have changed my implementation here: https://github.com/TwineHealth/grunt-api-benchmark/commit/ae65a9e13832acf37d6bcfcb0c5f01f6ccb982ec

**Comment by PascalPflaum** [comment]
@matteofigus 
This issue is quite old, but I'm facing the same problem. Would you merge a PR, if I write one based on sqilroys code? Of course extended with some documentation.

**Comment by matteofigus** [comment]
Yes absolutely, thanks @PascalPflaum
